scripts/validate_string_concatenation.py

```diff
@@ -1,51 +1,313 @@
+                )
+                break
+
+
+def strings_to_concatenate(
+    source_path: str,
+) -> Generator[Tuple[str, int, str], None, None]:
+    """
+    This test case is necessary after 'Black' (https://github.com/psf/black),
+    is formating strings over multiple lines.
+
+    For example, when this:
+
+    >>> foo = (
+    ...     "bar "
+    ...     "baz"
+    ... )
+
+    Is becoming this:
+
+    >>> foo = ("bar " "baz")
+
+    'Black' is not considering this as an
+    issue (see https://github.com/psf/black/issues/1051),
+    so we are checking it here instead.
+
+    Parameters
+    ----------
+    source_path : str
+        File path pointing to a single file.
+
+    Yields
+    ------
+    source_path : str
+        Source file path.
+    line_number : int
+        Line number of unconcatenated string.
+    msg : str
+        Explenation of the error.
+
+    Notes
+    -----
+    GH #30454
+    """
+    with open(source_path, "r") as file_name:
+        tokens: List = list(tokenize.generate_tokens(file_name.readline))
+
+    for current_token, next_token in zip(tokens, tokens[1:]):
+        if current_token.type == next_token.type == token.STRING:
+            yield (
+                source_path,
+                current_token.start[0],
+                (
+                    "String unnecessarily split in two by black. "
+                    "Please merge them manually."
+                ),
+            )
+
+
+def strings_with_wrong_placed_whitespace(
+    source_path: str,
+) -> Generator[Tuple[str, int, str], None, None]:
+    """
+    Test case for leading spaces in concated strings.
+
+    For example:
+
+    >>> rule = (
+    ...    "We want the space at the end of the line, "
+    ...    "not at the beginning"
+    ... )
+
+    Instead of:
+
+    >>> rule = (
+    ...    "We want the space at the end of the line,"
+    ...    " not at the beginning"
+    ... )
+
+    Parameters
+    ----------
+    source_path : str
+        File path pointing to a single file.
+
+    Yields
+    ------
+    source_path : str
+        Source file path.
+    line_number : int
+        Line number of unconcatenated string.
+    msg : str
+        Explenation of the error.
+    """
+
+    def get_prefix_len(line: str) -> int:
+        """
+        Get the prefix length from a string literal.
+
+        Parameters
+        ----------
+        line : str
+            Line to check.
+
+        Returns
+        -------
+        int
+            Length of the prefix of the string literal.
+
+        Examples
+        --------
+
+        >>> get_prefix_len('"Hello world"')
+        0
+
+        >>> get_prefix_len('f"Hello world"')
+        1
+
+        >>> get_prefix_len('rf"Hello world"')
+        2
+        """
+        for counter, character in enumerate(line):
+            if character == r'"' or character == r"'":
+                return counter
+        return 0
+
+    def whitespace_validation(first_line: str, second_line: str) -> bool:
```

Comment:
Since this returns True/False, I think it can make things clearer if the name of the function specifies what's True and what's False. For example `has_wrong_whitespaces`.